# Projeto - Fundamentos da Descoberta de Dados

Nesse projeto trabalharemos com a base de dados de produtos de um supermercado do Chile.
A premissa é aplicar os conceitos estatísticos e mais os conceitos de visualizações de dados através de gráficos


In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px

Iniciando a leitura dos dados do arquivo CSV com os dados do supermercado

In [2]:
df = pd.read_csv(r"Bases\MODULO7_PROJETOFINAL_BASE_SUPERMERCADO (1).csv", delimiter=',')

df.head(10)

,title,Marca,Preco_Normal,Preco_Desconto,Preco_Anterior,Desconto,Categoria
0,"Pack 12 un, Leche extra proteína 1 L",Loncoleche,19788,0,0,0,lacteos
1,"Pack 12 un, Leche chocolate receta original 1 L",Soprole,18228,0,0,0,lacteos
2,"Pack 12 un, Leche semidescremada chocolate 1 L",Soprole,18228,0,0,0,lacteos
3,"Pack 12 un, Leche semidescremada frutilla 1 L",Soprole,18228,0,0,0,lacteos
4,"Pack 12 un, Leche sin lactosa chocolate 1 L",Loncoleche,17988,0,0,0,lacteos
5,"Pack 12 un, Leche sin lactosa frutilla 1 L",Loncoleche,17988,0,0,0,lacteos
6,"Pack 12 un, Leche saborizada light chocolate 1 L",Loncoleche,17988,0,0,0,lacteos
7,"Pack 12 un, Leche saborizada frutilla 1 L",Colun,17388,0,0,0,lacteos
8,"Pack 12 un, Leche saborizada vainilla 1 L",Colun,17388,0,0,0,lacteos
9,"Pack 12 un, Leche saborizada manjar 1 L",Colun,17388,0,0,0,lacteos


Os campos do nosso dataframe são:

**Title:** Nome do produto.

**Marca:** A marca do produto.

**Preco_Normal:** O preço em que o produto costuma ser vendido quando não há desconto.

**Preco_Desconto:** O preço vendido após o desconto ser aplicado.

**Preco_Anterior:** Preço em que era comercializado o produto antes do desconto aplicado.

**Desconto:** Total de desconto aplicado.

**OBS**
- As colunas que aparecem com valores 0 são para os produtos onde não tivemos descontos aplicados.
- As categorias estão em espanhol!

### Retornando os dados da média e mediana dos preços dos produtos.

Serão retornados  realizado agrupamento de dados por categoria e será discriminado os valores que estiverem acima da média para fins validar valores


In [3]:
#Seu código aqui para a média
#Media
media_preco = df['Preco_Normal'].mean()
print(f"A média dos preços normais é: {media_preco}")

#Media pro categoria
df_media_categoria = df.groupby('Categoria')['Preco_Normal'].mean().reset_index().sort_values(by='Preco_Normal', ascending=False)

#criando uma nova coluna da informação se está a cima da media ou não
df_media_categoria['Acima_da_Media'] = df_media_categoria['Preco_Normal'] > media_preco

#mostando os dados
df_media_categoria = df_media_categoria.sort_values(by='Preco_Normal', ascending=False)
print(df_media_categoria)

A média dos preços normais é: 2070.9972899728996
                    Categoria  Preco_Normal  Acima_da_Media
1          comidas-preparadas   3095.043478            True
5                     lacteos   2385.219239            True
2                  congelados   2108.042553            True
0  belleza-y-cuidado-personal   1783.556485           False
3                      frutas   1724.473684           False
6                    verduras   1343.296875           False
4        instantaneos-y-sopas    765.491228           False


In [4]:
#Seu código aqui para a mediana
#Mediana
mediana_preco = df['Preco_Normal'].median()
print(f"A mediana dos preços normais é: {mediana_preco}")

#Media pro categoria
df_media_categoria = df.groupby('Categoria')['Preco_Normal'].mean().reset_index().sort_values(by='Preco_Normal', ascending=False)

#criando uma nova coluna  da informação se está a cima da media ou não
df_media_categoria['Acima_da_Mediana'] = df_media_categoria['Preco_Normal'] > mediana_preco

#mostando os dados
df_media_categoria = df_media_categoria.sort_values(by='Preco_Normal', ascending=False)
print(df_media_categoria)

A mediana dos preços normais é: 1269.0
                    Categoria  Preco_Normal  Acima_da_Mediana
1          comidas-preparadas   3095.043478              True
5                     lacteos   2385.219239              True
2                  congelados   2108.042553              True
0  belleza-y-cuidado-personal   1783.556485              True
3                      frutas   1724.473684              True
6                    verduras   1343.296875              True
4        instantaneos-y-sopas    765.491228             False


Avaliando os dados, as categorias que tem valor "médio" abaixo da "mediana" é apenas a categoria 'instantaneos-y-sopas'

As demais categorias apresentam preço "médio" superior a "mediana". São as categorias 
- comidas-preparadas
- lacteos
- congelados
- belleza-y-cuidado-personal
- frutas
- verduras

### Analisando o comportamento da média e da mediana para cada categoria em relação ao desvio padrão 

In [5]:
#Seu código aqui
df_desvio_padrao = df.groupby('Categoria')['Preco_Normal'].std().reset_index()
print(df_desvio_padrao)
#Criando o gráfico de barras

                    Categoria  Preco_Normal
0  belleza-y-cuidado-personal   2210.041719
1          comidas-preparadas   2019.911428
2                  congelados   2111.539896
3                      frutas   1639.151114
4        instantaneos-y-sopas   1170.232869
5                     lacteos   3925.816164
6                    verduras   1012.699625


Fazendo uma avaliação da dispersão dos valores em função da média, que apresenta o valor de 2070.99, notamos que as categorias de laticínios e verduras apresentam os maiores desvios.

A categoria laticínios apresenta o valor mais elevado(3925), seguidos de Beleza e cuidados pessoais(2210). Verduras(1012) com o menor valor, Seguidos de sopas e instantâneos(1170)



### Analisando a distribuição do Preco_Normal para a categoria com maior desvio padrão.

Como informado anteriormente, o maior desvio padrão foi identificado na categoria `lacteos` onde podemos ver no gráfico a seguir uma quantidade elevada de outliers

In [6]:
#Seu código aqui
fig = px.box(df.loc[df['Categoria'] == 'lacteos'], y='Preco_Normal', title='Distribuição de Preços Lacteos')
fig.show()


Avaliando o gráfico podemos identificar que existe uma grande variação nos preços, sugerindo que há produtos na amostra com preços muito diferentes. Podemos notar muitos valores acima do limite máximo o que podemos indicar que são outliers, sendo o maior valor apontado como 19788.


A mediana, apesar de apresentar um valor de 989, está mais próximo do primeiro quartil (269) o que pode indicar uma concentração maior de preços nesta faixa


### Gráfico de barras com a média de descontos por categoria.

In [7]:
# Seu código aqui
#Agrupando os dados por categoria e desconto médio
desconto_medio_categoria = df.groupby('Categoria')['Desconto'].mean().reset_index()

#Criando gráfico de barras 
fig = px.bar(desconto_medio_categoria, x='Desconto', y='Categoria', orientation='h',
              title='Desconto Médio por Categoria',
              labels={'Desconto': 'Desconto Médio', 'Categoria': 'Categoria'})
fig.show()

### Gráfico de mapa interativo agrupando os dados por categoria, marca e trazendo a média de desconto.

In [8]:
# Seu código aqui
#Agrupando os dados por categoria e desconto médio
desconto_medio_categoria = df.groupby(['Categoria', 'Marca'])['Desconto'].mean().reset_index()

fig = px.treemap(desconto_medio_categoria,
                 path=['Categoria', 'Marca'],
                 values='Desconto',
                 title='Media de desconto por marca e categoria',
                 color='Desconto')
# Exibindo o gráfico
fig.show()